tests/operator_test.go

```diff
@@ -275,7 +290,20 @@ var _ = Describe("Operator", func() {
 				return fmt.Errorf("Waiting for phase to be deployed")
 			}
 			return nil
-		}, 160*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}, time.Duration(timeoutSeconds)*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+	}
+
+	waitForKv := func(newKv *v1.KubeVirt) {
+		waitForKvWithTimeout(newKv, 300)
+	}
+
+	patchKvVersionAndRegistry := func(name string, version string, registry string) {
+		data := []byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/imageTag", "value": "%s"},{ "op": "replace", "path": "/spec/imageRegistry", "value": "%s"}]`, version, registry))
+		Eventually(func() error {
+			_, err := virtClient.KubeVirt(tests.KubeVirtInstallNamespace).Patch(name, types.JSONPatchType, data)
```

Comment:
So, to ensure I understand what's at stake here: This is looking in KubeVirt's installed namespace, so this test is actually patching the image's registry and tag that the operator itself is using to install KubeVirt?

If that's correct, what are the consequences if something goes wrong here? e.g. test is killed.

Reply:
we attempt to restore original kubevirt object in a AfterTest() function. What you're pointing out is a risk still though. It's not a new risk however. The operator test suite already does similar actions